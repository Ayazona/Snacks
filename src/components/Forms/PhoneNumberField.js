import React              from 'react'
import PropTypes          from 'prop-types'
import Radium             from 'radium'
import { colors }         from '../../styles'
import FormComponent      from './FormComponent'
import ValidationError    from './ValidationError'
import FloatingLabel      from './FloatingLabel'
import TextFieldHint      from './TextFieldHint'
import ServerError        from './ServerError'
import HelperText         from './HelperText'
import withTheme          from '../../styles/themer/withTheme'
import MaskedTextInput    from 'react-text-mask'
import { themePropTypes } from '../../styles/themer/utils'

const styles = {
  wrapper: {
    cursor: 'auto',
    display: 'inline-block',
    position: 'relative',
    width: '343px',
  },
  inputContainer: {
    borderRadius: '4px',
    position: 'relative'
  },
  input: {
    backgroundColor: '#FFF',
    border: `solid 1px ${colors.GRAY_74}`,
    borderRadius: '4px',
    boxSizing: 'border-box',
    color: colors.GRAY_20,
    fontSize: '16px',
    height: '56px',
    margin: '0',
    padding: '25px 8px 8px 8px',
    outline: 'none',
    position: 'relative',
    width: '100%',
    WebkitOpacity: 1,
    WebkitTapHighlightColor: 'rgba(0,0,0,0)'
  },
  inputDisabled: {
    border: `1px dashed ${colors.GRAY_74}`,
    backgroundColor: colors.GRAY_93,
    color: colors.GRAY_46,
    cursor: 'not-allowed'
  },
  inputError: {
    border: `1px solid ${colors.RED_700}`,
    backgroundColor: '#FDE6EB'
  },
  fullWidth: {
    width: '100%'
  },
  halfWidth: {
    width: '162px'
  }
}

const getSnackStyles = snacksTheme => {
  const { action } = snacksTheme.colors
  return {
    highlight: {
      border: `1px solid ${action}`
    }
  }
}

const getInputSyles = ({props, theme, isFocused}) => {
  const snacksStyles = getSnackStyles(theme)
  const { disabled, hasError, inputStyle } = props
  const disabledStlyes = disabled ? styles.inputDisabled : {}
  const errorStyles = !disabled && hasError ? styles.inputError : {}
  const focusedStyles = isFocused && !hasError ? snacksStyles.highlight : {}

  return {
    ...styles.input,
    ...inputStyle,
    ...disabledStlyes,
    ...errorStyles,
    ...focusedStyles
  }
}

@withTheme
@FormComponent
@Radium
class PhoneNumberField extends React.Component {
  static propTypes = {
    /** Name of the field */
    name               : PropTypes.string.isRequired,
    /** HTML autocomplete attribute */
    autoComplete       : PropTypes.string,
    /** DefaultValue for non controlled component */
    defaultValue       : PropTypes.any,
    /** Disable the text field */
    disabled           : PropTypes.bool,
    /** Text of label that will animate when TextField is focused */
    floatingLabelText  : PropTypes.string,
    /** Sets width to 100% */
    fullWidth          : PropTypes.bool,
    /** Sets width to 162px */
    halfWidth          : PropTypes.bool,
    /** FormComponent error for validation */
    hasError           : PropTypes.bool,
    /** Helper text will show up in bottom right corner below TextField */
    helperText         : PropTypes.string,
    /** Uniq id for input */
    id                 : PropTypes.string,
    /** Style for input */
    inputStyle         : PropTypes.object,
    /** Set by FormComponent by default.   */
    isValid            : PropTypes.bool,
    /** onFocus callback */
    onFocus            : PropTypes.func,
    /** onChange callback */
    onChange           : PropTypes.func,
    /** onBlur callback */
    onBlur             : PropTypes.func,
    /** onKeyDown callback */
    onKeyDown          : PropTypes.func,
    /** Mark the field as required.  */
    required           : PropTypes.bool,
    /** Error from server to show ServerError message */
    serverError        : PropTypes.string,
    /** Wrapper styles */
    style              : PropTypes.object,
    /** Text to show for validation error  */
    validationErrorText: PropTypes.string,
    /** Value will make TextField a controlled component  */
    value              : PropTypes.string,
    /** Snacks theme attributes provided by `Themer` */
    snacksTheme        : themePropTypes
  }

  static defaultProps = {
    autoComplete: 'on',
    disabled    : false,
    defaultValue: null,
    onKeyDown: () => {} // eslint-disable-line no-empty-function
  }

  state = {
    hasValue: this.props.defaultValue !== null || Boolean(this.props.value)
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.disabled && !this.props.disabled) {
      this.setState({ isFocused: false })
    }

    if (!this.state.hasValue && nextProps.value) {
      this.setState({hasValue: true})
    }
  }

  getValue = () => {
    if (!this.input) {
      return null
    }

    return this.input.value.replace(/\(|\)|-| /g, '')
  }

  handleInputChange = (e) => {
    const { onChange } = this.props
    const { hasValue } = this.state
    const value        = e.target.value
    // Limit setState call to only when hasValue changes
    if (value && !hasValue) {
      this.setState({hasValue: true})
    } else if (!value && hasValue) {
      this.setState({hasValue: false})
    }

    onChange && onChange(e, value.replace(/\(|\)|-| /g, ''), value)
  }

  handleInputFocus = (e) => {
    this.setState({isFocused: true})
    this.props.onFocus && this.props.onFocus(e)
  }

  handleInputBlur = (e) => {
    this.setState({isFocused: false})
    this.props.onBlur && this.props.onBlur(e)
  }

  handleKeyDown = (e) => {
    this.props.onKeyDown(e)
  }

  render() {
    const {
      floatingLabelText,
      defaultValue,
      disabled,
      fullWidth,
      halfWidth,
      hasError,
      id,
      isValid,
      name,
      required,
      serverError,
      validationErrorText,
      style,
      value,
      helperText,
      autoComplete,
      snacksTheme
    } = this.props

    const {
      hasValue,
      isFocused
    } = this.state

    const inputId = id

    return (
      <div
        style={[
          styles.wrapper,
          fullWidth && styles.fullWidth,
          halfWidth && styles.halfWidth,
          style
        ]}
      >

        {serverError && !disabled && !isValid &&
          <ServerError
            text={serverError}
          />
        }

        <div style={styles.inputContainer}>
          <FloatingLabel
            text={floatingLabelText}
            float={isFocused || hasValue}
            disabled={disabled}
            isActive={isFocused}
            hasError={hasError}
            htmlFor={inputId}
            style={{pointerEvents: 'none'}}
          />

          <TextFieldHint
            text={'(555) 555-555'}
            show={!hasValue && isFocused}
            disabled={disabled}
          />

          <MaskedTextInput
            mask={['(', /[1-9]/, /\d/, /\d/, ')', ' ', /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, /\d/]}
            id={inputId}
            guide={false}
            name={name}
            aria-required={required}
            aria-invalid={hasError}
            onBlur={this.handleInputBlur}
            onChange={this.handleInputChange}
            onFocus={this.handleInputFocus}
            onKeyDown={this.handleKeyDown}
            autoComplete={autoComplete}
            placeholder=""
            defaultValue={value !== undefined ? undefined : defaultValue}
            disabled={disabled}
            keepCharPositions={true}
            render={(ref, props) => (
              <input
                ref={(input) => {
                  this.input = input
                  ref(input)}
                }
                style={getInputSyles({
                  props: this.props,
                  theme: snacksTheme,
                  isFocused
                })}
                {...props}
              />
            )}
          />
        </div>

        <ValidationError
          text={validationErrorText}
          show={!disabled && !isValid && !serverError}
          inputId={inputId}
        />

        <HelperText helperText={helperText} />
      </div>
    )
  }
}

export default PhoneNumberField

