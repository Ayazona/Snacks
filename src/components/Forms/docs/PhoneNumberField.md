The PhoneNumberField is similar to the TextField component, but with input masking and validations for US phone numbers.

PhoneNumberField Example:

```js
<div style={{width: '335px'}}>
  <PhoneNumberField
    name="phone"
    floatingLabelText="Phone Number"
    halfWidth
    required
  />
</div>
```

In a future release, this component will also support phone numbers from outside the US.