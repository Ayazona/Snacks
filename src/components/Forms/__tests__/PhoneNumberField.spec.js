import React         from 'react'
import { StyleRoot } from 'radium'
import { mount, shallow } from 'enzyme'
import { spy }       from 'sinon'
import toJson        from 'enzyme-to-json'
import PhoneNumberField  from '../PhoneNumberField'

it('renders without error', () => {
  const mounted = mount(
    <StyleRoot>
      <div>
        <PhoneNumberField
          id="test_id"
          name="test"
          floatingLabelText="Phone Number"
          hintText="(555) 555-555"
        />
      </div>
    </StyleRoot>
  )

  expect(mounted).toBe.truthy
})

it('renders correctly', () => {
  const wrapper = mount(
    <StyleRoot>
      <div>
        <PhoneNumberField
          id="test_id"
          name="test"
          floatingLabelText="Phone Number"
          hintText="(555) 555-555"
        />
      </div>
    </StyleRoot>
  )
  
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders correctly with focus state', () => {
  const wrapper = mount(
    <StyleRoot>
      <div>
        <PhoneNumberField
          id="test_id"
          name="test"
          floatingLabelText="Phone Number"
          hintText="(555) 555-555"
        />
      </div>
    </StyleRoot>
  )

  wrapper.find('input').simulate('focus')
  
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('fires the onFocus prop', () => {
  const onFocus = spy()
  const wrapper = mount(
    <StyleRoot>
      <div>
        <PhoneNumberField
          id="test_id"
          name="test"
          floatingLabelText="Phone Number"
          hintText="(555) 555-555"
          onFocus={onFocus}
        />
      </div>
    </StyleRoot>
  )

  wrapper.find('input').simulate('focus')

  expect(onFocus.calledOnce).toBe(true)
})

it('fires the onBlur prop', () => {
  const onBlur = spy()
  const wrapper = mount(
    <StyleRoot>
      <div>
        <PhoneNumberField
          id="test_id"
          name="test"
          floatingLabelText="Phone Number"
          hintText="(555) 555-555"
          onBlur={onBlur}
        />
      </div>
    </StyleRoot>
  )

  wrapper.find('input').simulate('blur')

  expect(onBlur.calledOnce).toBe(true)
})

it('fires the onChange prop', () => {
  const onChange = spy()
  const wrapper = mount(
    <StyleRoot>
      <div>
        <PhoneNumberField
          id="test_id"
          name="test"
          floatingLabelText="Phone Number"
          hintText="(555) 555-555"
          onChange={onChange}
        />
      </div>
    </StyleRoot>
  )

  wrapper.find('input').simulate('change', {target: {value: 'My new value'}})

  expect(onChange.calledOnce).toBe(true)
})
