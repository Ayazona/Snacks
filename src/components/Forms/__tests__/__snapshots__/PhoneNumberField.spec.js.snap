// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<StyleRoot>
  <div
    data-radium={true}
  >
    <div
      key=".0"
    >
      <withTheme(FormComponent)
        floatingLabelText="Phone Number"
        hintText="(555) 555-555"
        id="test_id"
        name="test"
      >
        <FormComponent
          floatingLabelText="Phone Number"
          hintText="(555) 555-555"
          id="test_id"
          name="test"
          snacksTheme={
            Object {
              "colors": Object {
                "action": "#43B02A",
                "actionHover": "#177D05",
                "primaryBackground": "#FFFFFF",
                "primaryForeground": "#43B02A",
                "secondaryBackground": "#43B02A",
                "secondaryForeground": "#FFFFFF",
                "secondaryForegroundFocus": "#757575",
              },
            }
          }
        >
          <PhoneNumberField
            autoComplete="on"
            defaultValue={null}
            disabled={false}
            floatingLabelText="Phone Number"
            hasError={false}
            hintText="(555) 555-555"
            id="test_id"
            isValid={true}
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            serverError={null}
            snacksTheme={
              Object {
                "colors": Object {
                  "action": "#43B02A",
                  "actionHover": "#177D05",
                  "primaryBackground": "#FFFFFF",
                  "primaryForeground": "#43B02A",
                  "secondaryBackground": "#43B02A",
                  "secondaryForeground": "#FFFFFF",
                  "secondaryForegroundFocus": "#757575",
                },
              }
            }
          >
            <div
              data-radium={true}
              style={
                Object {
                  "cursor": "auto",
                  "display": "inline-block",
                  "position": "relative",
                  "width": 343,
                }
              }
            >
              <div
                data-radium={true}
                key=".1"
                style={
                  Object {
                    "borderRadius": 4,
                    "position": "relative",
                  }
                }
              >
                <withTheme(RadiumEnhancer)
                  disabled={false}
                  float={false}
                  hasError={false}
                  htmlFor="test_id"
                  key=".0"
                  style={
                    Object {
                      "pointerEvents": "none",
                    }
                  }
                  text="Phone Number"
                >
                  <FloatingLabel
                    disabled={false}
                    float={false}
                    hasError={false}
                    htmlFor="test_id"
                    snacksTheme={
                      Object {
                        "colors": Object {
                          "action": "#43B02A",
                          "actionHover": "#177D05",
                          "primaryBackground": "#FFFFFF",
                          "primaryForeground": "#43B02A",
                          "secondaryBackground": "#43B02A",
                          "secondaryForeground": "#FFFFFF",
                          "secondaryForegroundFocus": "#757575",
                        },
                      }
                    }
                    style={
                      Object {
                        "pointerEvents": "none",
                      }
                    }
                    text="Phone Number"
                  >
                    <label
                      data-radium={true}
                      htmlFor="test_id"
                      style={
                        Object {
                          "color": "#757575",
                          "cursor": "text",
                          "fontSize": "16px",
                          "margin": "0",
                          "pointerEvents": "none",
                          "position": "absolute",
                          "transform": "scale(1) translate(8px, 17px)",
                          "transformOrigin": "left top",
                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "userSelect": "none",
                          "zIndex": 1,
                        }
                      }
                    >
                      Phone Number
                    </label>
                  </FloatingLabel>
                </withTheme(RadiumEnhancer)>
                <TextFieldHint
                  disabled={false}
                  key=".1"
                  text="(555) 555-555"
                >
                  <div
                    data-radium={true}
                    style={
                      Object {
                        "color": "#BDBDBD",
                        "cursor": "inherit",
                        "fontSize": "16px",
                        "opacity": 0,
                        "pointerEvents": "none",
                        "position": "absolute",
                        "transform": "scale(1) translate(9px, 26px)",
                        "transformOrigin": "left top",
                        "transition": "opacity 250ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "zIndex": 1,
                      }
                    }
                  >
                    (555) 555-555
                  </div>
                </TextFieldHint>
                <t
                  aria-invalid={false}
                  autoComplete="on"
                  defaultValue={null}
                  disabled={false}
                  guide={false}
                  id="test_id"
                  keepCharPositions={true}
                  key=".2"
                  mask={
                    Array [
                      "(",
                      /\\[1-9\\]/,
                      /\\\\d/,
                      /\\\\d/,
                      ")",
                      " ",
                      /\\\\d/,
                      /\\\\d/,
                      /\\\\d/,
                      "-",
                      /\\\\d/,
                      /\\\\d/,
                      /\\\\d/,
                      /\\\\d/,
                    ]
                  }
                  name="test"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder=""
                  render={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="on"
                    defaultValue={null}
                    disabled={false}
                    id="test_id"
                    name="test"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    style={
                      Object {
                        "WebkitOpacity": 1,
                        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                        "backgroundColor": "#FFFFFF",
                        "border": "solid 1px #BDBDBD",
                        "borderRadius": 4,
                        "boxSizing": "border-box",
                        "color": "#424242",
                        "fontSize": 16,
                        "height": 56,
                        "margin": "0",
                        "outline": "none",
                        "padding": "25px 8px 8px 8px",
                        "position": "relative",
                        "width": "100%",
                      }
                    }
                  />
                </t>
              </div>
              <ValidationError
                inputId="test_id"
                key=".2"
                show={false}
              >
                <div
                  aria-atomic={true}
                  aria-live="assertive"
                  data-radium={true}
                  id="error_test_id"
                  style={
                    Object {
                      "color": "#B30029",
                      "display": "none",
                      "fontSize": "12px",
                      "marginLeft": "1px",
                      "marginTop": "2px",
                      "opacity": "0",
                      "pointerEvents": "none",
                      "textAlign": "left",
                      "transition": "opacity 250ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    }
                  }
                />
              </ValidationError>
              <HelperText
                key=".3"
              />
            </div>
          </PhoneNumberField>
        </FormComponent>
      </withTheme(FormComponent)>
    </div>
    <StyleSheet
      key=".1"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </StyleSheet>
  </div>
</StyleRoot>
`;

exports[`renders correctly with focus state 1`] = `
<StyleRoot>
  <div
    data-radium={true}
  >
    <div
      key=".0"
    >
      <withTheme(FormComponent)
        floatingLabelText="Phone Number"
        hintText="(555) 555-555"
        id="test_id"
        name="test"
      >
        <FormComponent
          floatingLabelText="Phone Number"
          hintText="(555) 555-555"
          id="test_id"
          name="test"
          snacksTheme={
            Object {
              "colors": Object {
                "action": "#43B02A",
                "actionHover": "#177D05",
                "primaryBackground": "#FFFFFF",
                "primaryForeground": "#43B02A",
                "secondaryBackground": "#43B02A",
                "secondaryForeground": "#FFFFFF",
                "secondaryForegroundFocus": "#757575",
              },
            }
          }
        >
          <PhoneNumberField
            autoComplete="on"
            defaultValue={null}
            disabled={false}
            floatingLabelText="Phone Number"
            hasError={false}
            hintText="(555) 555-555"
            id="test_id"
            isValid={true}
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            serverError={null}
            snacksTheme={
              Object {
                "colors": Object {
                  "action": "#43B02A",
                  "actionHover": "#177D05",
                  "primaryBackground": "#FFFFFF",
                  "primaryForeground": "#43B02A",
                  "secondaryBackground": "#43B02A",
                  "secondaryForeground": "#FFFFFF",
                  "secondaryForegroundFocus": "#757575",
                },
              }
            }
          >
            <div
              data-radium={true}
              style={
                Object {
                  "cursor": "auto",
                  "display": "inline-block",
                  "position": "relative",
                  "width": 343,
                }
              }
            >
              <div
                data-radium={true}
                key=".1"
                style={
                  Object {
                    "borderRadius": 4,
                    "position": "relative",
                  }
                }
              >
                <withTheme(RadiumEnhancer)
                  disabled={false}
                  float={true}
                  hasError={false}
                  htmlFor="test_id"
                  isActive={true}
                  key=".0"
                  style={
                    Object {
                      "pointerEvents": "none",
                    }
                  }
                  text="Phone Number"
                >
                  <FloatingLabel
                    disabled={false}
                    float={true}
                    hasError={false}
                    htmlFor="test_id"
                    isActive={true}
                    snacksTheme={
                      Object {
                        "colors": Object {
                          "action": "#43B02A",
                          "actionHover": "#177D05",
                          "primaryBackground": "#FFFFFF",
                          "primaryForeground": "#43B02A",
                          "secondaryBackground": "#43B02A",
                          "secondaryForeground": "#FFFFFF",
                          "secondaryForegroundFocus": "#757575",
                        },
                      }
                    }
                    style={
                      Object {
                        "pointerEvents": "none",
                      }
                    }
                    text="Phone Number"
                  >
                    <label
                      data-radium={true}
                      htmlFor="test_id"
                      style={
                        Object {
                          "color": "#43B02A",
                          "cursor": "inherit",
                          "fontSize": "16px",
                          "margin": "0",
                          "pointerEvents": "none",
                          "position": "absolute",
                          "transform": "scale(0.85) translate(8px, 6px)",
                          "transformOrigin": "left top",
                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "userSelect": "none",
                          "zIndex": 1,
                        }
                      }
                    >
                      Phone Number
                    </label>
                  </FloatingLabel>
                </withTheme(RadiumEnhancer)>
                <TextFieldHint
                  disabled={false}
                  key=".1"
                  show={true}
                  text="(555) 555-555"
                >
                  <div
                    data-radium={true}
                    style={
                      Object {
                        "color": "#BDBDBD",
                        "cursor": "inherit",
                        "fontSize": "16px",
                        "opacity": 1,
                        "pointerEvents": "none",
                        "position": "absolute",
                        "transform": "scale(1) translate(9px, 26px)",
                        "transformOrigin": "left top",
                        "transition": "opacity 250ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "zIndex": 1,
                      }
                    }
                  >
                    (555) 555-555
                  </div>
                </TextFieldHint>
                <t
                  aria-invalid={false}
                  autoComplete="on"
                  defaultValue={null}
                  disabled={false}
                  guide={false}
                  id="test_id"
                  keepCharPositions={true}
                  key=".2"
                  mask={
                    Array [
                      "(",
                      /\\[1-9\\]/,
                      /\\\\d/,
                      /\\\\d/,
                      ")",
                      " ",
                      /\\\\d/,
                      /\\\\d/,
                      /\\\\d/,
                      "-",
                      /\\\\d/,
                      /\\\\d/,
                      /\\\\d/,
                      /\\\\d/,
                    ]
                  }
                  name="test"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder=""
                  render={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="on"
                    defaultValue={null}
                    disabled={false}
                    id="test_id"
                    name="test"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder=""
                    style={
                      Object {
                        "WebkitOpacity": 1,
                        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                        "backgroundColor": "#FFFFFF",
                        "border": "1px solid #43B02A",
                        "borderRadius": 4,
                        "boxSizing": "border-box",
                        "color": "#424242",
                        "fontSize": 16,
                        "height": 56,
                        "margin": "0",
                        "outline": "none",
                        "padding": "25px 8px 8px 8px",
                        "position": "relative",
                        "width": "100%",
                      }
                    }
                  />
                </t>
              </div>
              <ValidationError
                inputId="test_id"
                key=".2"
                show={false}
              >
                <div
                  aria-atomic={true}
                  aria-live="assertive"
                  data-radium={true}
                  id="error_test_id"
                  style={
                    Object {
                      "color": "#B30029",
                      "display": "none",
                      "fontSize": "12px",
                      "marginLeft": "1px",
                      "marginTop": "2px",
                      "opacity": "0",
                      "pointerEvents": "none",
                      "textAlign": "left",
                      "transition": "opacity 250ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    }
                  }
                />
              </ValidationError>
              <HelperText
                key=".3"
              />
            </div>
          </PhoneNumberField>
        </FormComponent>
      </withTheme(FormComponent)>
    </div>
    <StyleSheet
      key=".1"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </StyleSheet>
  </div>
</StyleRoot>
`;
